apps/rush-lib/src/cli/scriptActions/PhasedScriptAction.ts

```diff
@@ -116,6 +117,9 @@ export class PhasedScriptAction extends BaseScriptAction<IPhasedCommand> {
     const terminal: Terminal = new Terminal(this.rushSession.terminalProvider);
     let buildCacheConfiguration: BuildCacheConfiguration | undefined;
     if (!this._disableBuildCache) {
+      await this.parser.pluginManager.hooks.initializeFeatureKind
+        .get(FeatureKind.buildCacheProvider)
+        ?.promise();
       buildCacheConfiguration = await BuildCacheConfiguration.tryLoadAsync(
         terminal,
         this.rushConfiguration,
```

Comment:
I'd expect this to look more like:
```ts
const buildCacheProviderHooks: BuildCacheProviderHooks = await this.parser.pluginManager.initializeAsync(FeatureKind.buildCacheProvider);
buildCacheConfiguration  = await BuildCacheConfiguration.tryLoadAsync(terminal, this.rushConfiguration, buildCacheProviderHooks);
```

Accessing the APIs that are relevant to any given `FeatureKind` should require handing the plugin infrastructure the `FeatureKind`. The plugin requests the relevant API surface by handing over the `FeatureKind`, and the internals of Rush that support plugins hand the plugin manager the `FeatureKind` to get the fully-populated hooks.